be/src/vec/exprs/vmulti_match_predicate.cpp

```diff
@@ -0,0 +1,230 @@
+    if (scope == FunctionContext::THREAD_LOCAL || scope == FunctionContext::FRAGMENT_LOCAL) {
+        context->fn_context(_fn_context_index)->set_function_state(scope, _inverted_index_ctx);
+    }
+    if (scope == FunctionContext::FRAGMENT_LOCAL) {
+        RETURN_IF_ERROR(VExpr::get_const_col(context, nullptr));
+    }
+    _open_finished = true;
+    return Status::OK();
+}
+
+void VMultiMatchPredicate::close(VExprContext* context, FunctionContext::FunctionStateScope scope) {
+    VExpr::close_function_context(context, scope, _function);
+    VExpr::close(context, scope);
+}
+
+Status VMultiMatchPredicate::evaluate_inverted_index(VExprContext* context,
+                                                     uint32_t segment_num_rows) const {
+    auto children_num = get_num_children();
+    DCHECK_GE(children_num, 3);
+    vectorized::ColumnsWithTypeAndName arguments;
+    segment_v2::InvertedIndexResultBitmap ret;
+    // last argument is query value
+    if (get_child(children_num - 1)->is_literal()) {
+        auto* column_literal = assert_cast<VLiteral*>(get_child(children_num - 1).get());
+        arguments.emplace_back(column_literal->get_column_ptr(), column_literal->get_data_type(),
+                               column_literal->expr_name());
+    } else {
+        return Status::NotSupported(
+                "child {} in evaluate_inverted_index for VMultiMatchPredicate must be "
+                "literal, but we got {}",
+                children_num - 1, get_child(children_num - 1)->expr_name());
+    }
+    // second last argument is function name
+    if (get_child(children_num - 2)->is_literal()) {
+        auto* column_literal = assert_cast<VLiteral*>(get_child(children_num - 2).get());
+        auto match_type = column_literal->value();
+        if (match_type != "phrase_prefix") {
+            return Status::NotSupported("query type is incorrect, only support phrase_prefix");
+        }
+    } else {
+        return Status::NotSupported(
+                "child {} in evaluate_inverted_index for VMultiMatchPredicate must be "
+                "literal, but we got {}",
+                children_num - 1, get_child(children_num - 1)->expr_name());
+    }
+    std::set<int> column_ids;
+    for (int child_num = 0; child_num < children_num - 2; child_num++) {
+        if (get_child(child_num)->is_slot_ref()) {
+            auto* column_slot_ref = assert_cast<VSlotRef*>(get_child(child_num).get());
+            column_ids.insert(column_slot_ref->column_id());
+        } else {
+            return Status::NotSupported(
+                    "child {} in evaluate_inverted_index for VMultiMatchPredicate must be slot "
+                    "ref, but "
+                    "we "
+                    "got {}",
+                    child_num, get_child(child_num)->expr_name());
+        }
+    }
+
+    for (const auto& col_id : column_ids) {
+        auto result_bitmap = DORIS_TRY(
```

Comment:
If one of the multi_match queries fails, is it possible to fallback to a non-indexed match query?